code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -130,6 +136,20 @@
 				pic = new('icons/turf/overlays.dmi', place, "redOverlay", AREA_LAYER)
 			flick_overlay(pic, list(user.client), 8)
 
+/obj/item/clothing/glasses/meson/engine/proc/show_connections()
+	var/mob/living/carbon/human/user = loc
+	var/list/connection_images = list()
+	for(var/obj/machinery/atmospherics/pipe/smart/smart in orange(range, user))
+		for(var/direction in GLOB.cardinals)
+			var/image/connection
+			if(!(smart.GetInitDirections() & direction))
```

Comment:
This should probably be outside the loop. Also could possible do not regenerate these every show_connection call if directions didn't change.

Reply:
i tried to make the image cached, but you'd still have to make new of them for every pipe in the view, so it would still need to be made each time, if there is another way i can't see it right now

Reply:
What i meant by caching is keep it between show_connections calls. Make connection_images indexed by pipe/ref + their dir. If that didn't change no need to redo anything.

Reply:
i think i did it, lemme know if is good or the wrong way